Code/Source/RGLSystemComponent.cpp

```diff
@@ -189,6 +232,26 @@ namespace RGL
         m_modelLibrary.Clear();
     }
 
+    void RGLSystemComponent::OnTick(float deltaTime, AZ::ScriptTimePoint time)
+    {
+        for (auto entityId : m_managersToBeRemoved)
+        {
+            m_entityManagers.erase(entityId);
+            m_unprocessedEntities.insert(entityId);
+        }
+        m_managersToBeRemoved.clear();
+    }
```

Comment:
Why couldn't we erase entity manager in `ReviseEntityPresence(AZ::EntityId entityId)`?

Reply:
This is so that the caller can signal revision inside a bus notification and can then safely disconnect from said bus after the call has ended (on destruction).

Reply:
At the moment, the only caller is `EntityTagListener` which is not destroyed in `ReviseEntityPresence()` so your case doesn't occur. Is it solution for the future usages?

Reply:
Exactly. I believe that this could lead to bugs that would be hard to track down as in previous iterations of the solution the entity manager handlet the tag notification bus and the `ReviseEntityPresence` method was called from within the callbacks which lead to segfaults caused by destruction of the entity manager before the callback was finished. In my opinion a small amount of additional complexity in this case is worth it.